### DISCUSSION TITLE
Streaming large amounts of metrics results in gaps in graphs

### DISCUSSION THREAD

**Discussion by mjksohonet** [netdata/netdata#15241] [Q&A]
Hi,
I have a number of XCP-ng 8.2 servers using netdata 1.19 to collect metrics.
As netdata 1.19 is unable to export metrics directly to graphite, I have built a netdata 1.40 parent server and am streaming the metrics to it.
What I have noticed is that when the number of metrics on any node goes the low thousands, gaps start appreaing in the graphs for that node when viewed on the parentnode and it looks like those metrics are not being exported to graphite as well.
How do I go about troubleshooting this?

**Comment by ilyam8** [comment]
Hi, @mjksohonet. I think it is hard to answer without details but general advice is:

- update both parent and children instances to the latest stable.
- give more memory and CPU resources to the parent instance if needed.

Additionally, it is important to check if the network bandwidth is sufficient.

For instance, we have a lab with 240+ children. Every child streams ~5.5K. It makes it more than 1 million per second. No gaps, 180 megabits/s (can be much more after the parent restart because of replication). Netdata uses 8-10 CPU cores (so much CPU because the instance restreams all data to another parent, does ML and health. Can be much more after the parent restart because of replication, Netdata will try to utilize all available resources).

So one parent instance scales very well if you give it enough resources.

And children should not use the `none` memory mode, but at least `ram` - to not lose metrics during periods when they cannot stream the parent (parent restart, networking issues, etc.). All the gaps will be replicated on reconnection during replication.
